### ISSUE TITLE
Website generator

### ISSUE THREAD

**Issue by lwjohnst86** [UofTCoders/grad-course#7]
Here are some options for generating the website:

- Using the R package [blogdown](https://bookdown.org/yihui/blogdown/), which I've started using for a [course](https://dda-rcourse.lwjohnst.com/) I am running here in Denmark.
    - Pros: Really nice themes and layout, surprisingly easy to set up and use, can include python code chunks. Uses [Hugo](https://gohugo.io/) which is a pretty powerful static website generator.
    - Cons: It's mainly around R and R Markdown, which maybe most Python users have no idea about or how to use. 
- Use Hugo straight up, without blogdown and link to Python Notebook generated HTML files, so we can by pass the use of blogdown.
- Use Jekyll straight up, and like Hugo above, link the Notebooks directly. Con: Jekyll in my opinion is a bit more finnicky to use then Hugo.

Any other options/suggestions?

**Comment by joelostblom** [comment]
I also think Hugo is quite nice, and fast! There is a package to integrate
jupyter notebooks with Hugo, I have not used it myself. There is also
Nikola which supports notebooks natively.
https://mglerner.github.io/posts/switching-to-nikloa-for-jupyter-notebooks-and-a-static-site.html

**Comment by linamnt** [comment]
I hadn't heard of Hugo but having looked it up seems great and lots of
features, +1 for jupyter notebook integration. Seems to be growing so
there's likely more support and documentation for long term? Or I'd think.

**Comment by joelostblom** [comment]
Yes, it is pretty much as popular as Jekyll, at least in GitHub stats https://www.staticgen.com/.

This is the notebook integration https://github.com/knowsuchagency/hugo_jupyter

**Comment by lwjohnst86** [comment]
Awesome! Sounds like we have a general agreement about using Hugo then! Plus, if there are at least two of us who've used Hugo before (I assume you are meaning you've used it right Joel?), then it will be much much easier and less time consuming to create the website.
